euclidigons.lua

```diff
@@ -52,6 +52,11 @@ for v = 1, num_voices do
 	voice_manager.style.slots[v].on_release = voice_release
 	voice_manager.style.slots[v].on_steal = voice_remove
 end
+function generate_scale(root_num, scale_type)
+	local result = musicutil.generate_scale(root_num, scale_type, 1)
+	table.remove(result, #result)
+	return result
+end
```

Comment:
The reason we need our own function is that `musicutil.generate_scale` generates the root note twice. The closing note of the scale is the root note an octave higher.

Without this function the root note occupied two slots: first as the last note of one octave, and then immediately as the first note of the next octave. This meant that when using the built-in encoder, you had to travel twice the distance to change the root note to a different one.

This was easy to miss in the original script when the built-in Norns encoders were used but is obvious on Arc in the presence of visual notches on the note encoder. Then you see that there are two values assigned to the same note, which is unexpected.

This function removes the last entry and so the changing the notes with Arc or with the built-in encoder progresses linearly.